### DISCUSSION TITLE
Can't set `<PublishAot>` in the .csproj file with multiple `TargetFrameworks`

### DISCUSSION THREAD

**Discussion by felixf4xu** [unoplatform/uno#18843] [Q&A] [answered]
### Current behavior

In a Uno project with multiple `TargetFrameworks`, I can't set `<PublishAot>` to enable AOT, the error message is:
> error NETSDK1203: Ahead-of-time compilation is not supported for the target runtime identifier 'browser-wasm'.

### Expected behavior

Standard AOT configurations should be able to use in Uno project file, like:
```
      <PublishTrimmed>true</PublishTrimmed>
      <PublishAot>true</PublishAot>
      <PublishSingleFile>true</PublishSingleFile>
```

### How to reproduce it (as minimally and precisely as possible)

A project file with multiple TargetFrameworks:

```
<TargetFrameworks>net9.0-android;net9.0-ios;net9.0-maccatalyst;net9.0-windows10.0.26100;net9.0-browserwasm;net9.0-desktop</TargetFrameworks>
```
Add these configurations into the .csproj file
```
      <PublishTrimmed>true</PublishTrimmed>
      <PublishAot>true</PublishAot>
      <PublishSingleFile>true</PublishSingleFile>
```

then run `dotnet publish -f net9.0-desktop`

### Workaround

I have to remove `net9.0-browserwasm` from the `<TargetFrameworks>` temporarily to run:
```sh
dotnet publish -f net9.0-desktop
```

### Works on UWP/WinUI

None

### IDE

Visual Studio 2022

### Anything else we need to know?

Same error on Ubuntu if only `dotnet publish -f net9.0-desktop` is used (no VS publish)

**Comment by jeromelaban** [comment, marked as answer]
Thanks for the report. .NET WebAssembly does not support AOT to be set this way, so does .NET Android.

If you want to use those properties, you'll need to set them in a TFM conditional way, as such:

```xml
<PropertyGroup Condition=" '$(TargetFramework)' == 'net9.0-desktop' ">
      <PublishTrimmed>true</PublishTrimmed>
      <PublishAot>true</PublishAot>
      <PublishSingleFile>true</PublishSingleFile>
</PropertyGroup>
```

Note that single file publishing is not yet fully supported by Uno, particularly on Windows.
